tools/extras/check_dependencies.sh

```diff
@@ -87,28 +87,40 @@ if ! which awk >&/dev/null; then
   add_packages gawk gawk gawk
 fi
 
-if which python >&/dev/null ; then
+pythonok=true
+if ! which python2.7 >&/dev/null; then
+  echo "$0: python2.7 is not installed"
+  add_packages python2.7
+  pythonok=false
+fi
+
+if ! which python3 >&/dev/null; then
+  echo "$0: python3 is not installed"
+  add_packages python3
+  pythonok=false
+fi
+
+( 
+#Use a subshell so that sourcing env.sh does not have an influence on the rest of the script
+[ -f ./env.sh ] && . ./env.sh
+if $pythonok && ! which python2 >&/dev/null; then
+  mkdir -p $PWD/python
+  echo "$0: python2.7 is installed, but the python2 binary does not exist. Adding this to tools/env.sh"
+  ln -s $(which python2.7) $PWD/python/python2
+  echo "export PATH=$PWD/python:\${PATH}" >> env.sh
+fi
+
+if $pythonok && which python >&/dev/null && [[ ! -f $PWD/python/stubborn ]]; then
   version=`python 2>&1 --version | awk '{print $2}' `
   if [[ $version != "2.7"* ]] ; then
-    if which python2.7 >&/dev/null  || which python2 >&/dev/null ; then
-      echo "$0: python 2.7 is not the default python. You should either make it"
-      echo "$0: default or create an bash alias for kaldi scripts to run correctly"
-      status=1
-    else
-      echo "$0: python 2.7 is not installed"
-      add_packages python2.7 python python2.7
-    fi
-  fi
-else
-  if which python2.7 >&/dev/null  || which python2 >&/dev/null ; then
-    echo "$0: python 2.7 is not the default python. You should either make it"
-    echo "$0: default or create an bash alias for kaldi scripts to run correctly"
-    status=1
-  else
-    echo "$0: python is not installed (we need python 2.7)"
-    add_packages python2.7 python python2.7
+    echo "$0: WARNING python 2.7 is not the default python. We fixed this by adding a correct symlink more prominently on the path."
+    echo "$0: If you really want to use python3 as default, remove $PWD/python/python and add an empty file $PWD/python/stubborn"  
```

Comment:
this is a confusing message imho.
creating that stubborn file does not really imply using python3 -- it implies using whatever is linked to "python"

Reply:
You are absolutely right. I created the "stubborn" hook for some testing locally. I'm thinking I could remove that, as those that would want to be stubborn can just edit this script themselves. Does that sound better?